src/main/java/estaruma/cupcake/BakeryItem.java

```diff
@@ -0,0 +1,22 @@
+package estaruma.cupcake;
+
+// base for all cakes and toppings
+public class BakeryItem {
```

Comment:
You could have a Cake interface with methods for getting Price and Name, and 
a separate abstract class called CakeDecorator that implements Cake interface, which can add additional behaviours to the Cake?  

something like 

```
interface Cake {
    String getName();
    double getPrice();
}
```

```
abstract class CakeDecorator implements Cake {
    protected Cake cake;

    public CakeDecorator(Cake cake) {
        this.cake = cake;
    }

    @Override
    public String getName() {
        return cake.getName();
    }

    @Override
    public double getPrice() {
        return cake.getPrice();
    }
}
```